package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -0,0 +1,201 @@
+selected frames or whole trajectories.
+
+A list of time steps that contain phi and psi angles for each residue is
+generated, and a basic Ramachandran plot can be generated using the attribute
+:meth:`Ramachandran.plot`. This plot is best used as a reference, but it also
+allows for user customization.
+
+
+See Also
+--------
+:mod:`MDAnalysis.lib.distances`
+   contains functions to calculate dihedral angles from atom positions
+
+
+Example application
+-------------------
+This example will show how to calculate the phi and psi angles of adenylate
+kinase and generate a basic Ramachandran plot. The trajectory is included with
+the test data files::
+
+   import MDAnalysis as mda
+   from MDAnalysisTests.datafiles import GRO, XTC
+   u = mda.Universe(GRO, XTC)
+   r = u.select_atoms("protein")    # selection of residues
+
+   from MDAnalysis.analysis.dihedrals import Ramachandran
+   R = Ramachandran(r).run()
+
+   import matplotlib.pyplot as plt
+   fig, ax = plt.subplots(figsize=plt.figaspect(1))
+   ax.set_title("Ramachandran Plot (AdK)")
+   R.plot(ax=ax, color='k', marker='s')
+
+Alternatively, if you wanted to plot the data yourself, the angles themselves
+can be accessed using :meth:`Ramachandran.angles`::
+
+   fig, ax = plt.subplots(figsize=plt.figaspect(1))
+   ax.axis([-180,180,-180,180])
+   ax.axhline(0, color='k', lw=1)
+   ax.axvline(0, color='k', lw=1)
+   ax.set(xticks=range(-180,181,60), yticks=range(-180,181,60),
+          xlabel=r"$\phi$ (deg)", ylabel=r"$\psi$ (deg)")
+   for ts in R.angles:
+       ax.scatter(ts[:,0], ts[:,1], color='k', marker='s')
+
+Analysis Class
+--------------
+
+.. autoclass:: Ramachandran
+   :members:
+   :inherited-members:
+
+   .. attribute:: angles
+
+       Contains the time steps of the phi and psi angles for each residue as
+       an n_frames×n_residuesx2 :class:`numpy.ndarray` array with content
+       ``[[[phi, psi], [residue 2], ...], [time step 2], ...]``.
+
+"""
+from __future__ import absolute_import
+from six.moves import zip, range
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+import warnings
+
+import MDAnalysis as mda
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.lib.distances import calc_dihedrals
+
+
+class Ramachandran(AnalysisBase):
+    r"""Calculate phi and psi dihedral angles of specified residues.
+
+    Phi and psi angles wil be calculated for each residue in `atomgroup` for
+    each time step in the trajectory. A ReisdueGroup is generated from
+    `atomgroup` which is compared to the protein to determine if it is a
+    legitimate selection.
+
+    Note
+    ----
+    If the residue selection is beyond the scope of the protein, then an error
+    will be raised. If the residue selection includes the first or last residue
+    then a warning will be raised and they will be removed from the list of
+    residues, but the analysis will still run.
+
+    Run the analysis with :meth:`Ramachandran.run()`, whcih stores the results in the
```

Comment:
which

Reply:
Thanks I missed that. I'll probably go through the docstring one last time to catch anything else like that.